x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/SpatialAggregateFunction.java

```diff
@@ -22,26 +25,34 @@
  * select the best one.
  */
 public abstract class SpatialAggregateFunction extends AggregateFunction {
-    protected final boolean useDocValues;
+    protected final FieldExtractPreference preference;
 
-    protected SpatialAggregateFunction(Source source, Expression field, Expression filter, boolean useDocValues) {
+    protected SpatialAggregateFunction(Source source, Expression field, Expression filter, FieldExtractPreference values) {
         super(source, field, filter, emptyList());
-        this.useDocValues = useDocValues;
+        this.preference = values;
     }
 
-    protected SpatialAggregateFunction(StreamInput in, boolean useDocValues) throws IOException {
+    protected SpatialAggregateFunction(StreamInput in, FieldExtractPreference preference) throws IOException {
         super(in);
         // The useDocValues field is only used on data nodes local planning, and therefor never serialized
-        this.useDocValues = useDocValues;
+        this.preference = preference;
     }
 
     public abstract SpatialAggregateFunction withDocValues();
 
+    @Override
+    public boolean checkLicense(XPackLicenseState state) {
```

Comment:
Are we testing this anywhere?

Reply:
Could you please direct me to an example license test?

Reply:
`CheckLicenseTests` tests the licensing checks in general. I wonder if there is a way to write a negative test for our code? We could look in the SpatialPlugin for similar tests.